src/prefect/tasks.py

```diff
@@ -1284,7 +1287,8 @@ def delay(self, *args: P.args, **kwargs: P.kwargs) -> PrefectDistributedFuture:
         """
         return self.apply_async(args=args, kwargs=kwargs)
 
-    def serve(self) -> "Task":
+    @sync_compatible
+    async def serve(self):
```

Comment:
fwiw, i've been using `typing.NoReturn` for things that never return
```suggestion
    async def serve(self) -> NoReturn:
```

Reply:
Nice!